### ISSUE TITLE
Bug: Color Bulb can't change color and is not dimmable

### ISSUE THREAD

**Issue by smartmatic** [OpenWonderLabs/homebridge-switchbot#97]
### Describe The Bug

Basically i cant siwtch on and off the bulb in Apple Home. But i cant change the color. The color wheel is not available. If i turn the displayed white wheel nothing happens. In addition the bulb does not dim up or down if i try to adjust it.

### To Reproduce

Add Bulb and try it

### Expected behavior

To be able to change the color with standard color wheel and to dim the light up and down

### Relevant log output

```shell
[11/17/2021, 8:47:27 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command
[11/17/2021, 8:47:28 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command
[11/17/2021, 8:47:28 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command

[11/17/2021, 8:47:59 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command
[11/17/2021, 8:47:59 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command
[11/17/2021, 8:48:00 PM] [SwitchBot] Sending request for Color Bulb DD5A to SwitchBot API. command: turnOn parameter: default commandType: command
```

### Config for homebridge-switchbot

```shell
{
            "name": "SwitchBot",
            "credentials": {
                "openToken": "myToken",
                "notice": "Keep your token a secret!"
            },
            "options": {
                "refreshRate": 120
            },
            "devicediscovery": true,
            "platform": "SwitchBot"
        },
```

### Screenshots

![Switchbot_Color_Wheel - 1](https://user-images.githubusercontent.com/74932663/142272825-249f45f5-0237-4dd8-8ef5-a0eaa0d92fae.jpeg)

![Switchbot_Color_Bulb_Slider - 1](https://user-images.githubusercontent.com/74932663/142272841-dd4c2a1b-0bf3-48aa-ae85-11a38f2e9143.jpeg)

### Device & Model

SwitchBot Color Bulb

### Node.js Version

v14.18.0

### NPM Version

v7.24.1

### Homebridge Version

v1.3.6

### Homebridge Switchbot Plugin Version

v.1.1.0

### Homebridge Config UI X Plugin Version

v4.41.2

### Operating System

Raspbian

**Comment by donavanbecker** [comment]
@smartmatic do it works, but doesn't do color or dimming?

can you enable device logs and post them? So I can see more details?

**Comment by smartmatic** [comment]
I had some trouble with my Homebridge environment and set up a new one. These running now with Node v16.13.0 and NPM v8.1.0. The other settings are still the same as mentioned in my initial post.

Here is now the device Log for the Bulb.
[homebridge.log-3.txt](https://github.com/OpenWonderLabs/homebridge-switchbot/files/7561905/homebridge.log-3.txt)

Although the log sends a success message neither the brighness nor the color is changing. 
Hope the log helps?

**Comment by donavanbecker** [comment]
@smartmatic Thanks this helps. I will look into this.

**Comment by donavanbecker** [comment]
@smartmatic , can you try Beta, and get me the log that will output in yellow when you make a change in HomeKit?: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=1.1.1-beta.9/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by donavanbecker** [comment]
@smartmatic , can you try Beta, and get me the log that will output in yellow when you make a change in HomeKit?: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=1.1.1-beta.10/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by smartmatic** [comment]
> @smartmatic , can you try Beta, and get me the log that will output in yellow when you make a change in HomeKit?:
> 
> [![npm version](https://camo.githubusercontent.com/51a00cd3b9fb5dd88e204152402cd002fa5232f47ec64ca6726fb47ce1b23662/68747470733a2f2f62616467656e2e6e65742f62616467652f686f6d656272696467652d737769746368626f7440626574612f2533453d312e312e312d626574612e31302f6f72616e6765)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)
> 
> [Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

Thank you! I think i can test it today and send you feedback as soon as possible.

**Comment by donavanbecker** [comment]
> Thank you! I think i can test it today and send you feedback as soon as possible.

sounds good! it will actually be `beta.11` now.

**Comment by smartmatic** [comment]
I turned on Debug logging. Immediately after reboot of Homebridge i get the following error

`[19.11.2021, 17:38:09] [SwitchBot] Failed to Discover Devices. "Request failed with status code 401"
[19.11.2021, 17:38:09] [SwitchBot] [DEBUG] {"message":"Request failed with status code 401","name":"Error","stack":"Error: Request failed with status code 401\n    at createError (/usr/local/lib/node_modules/@switchbot/homebridge-switchbot/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/usr/local/lib/node_modules/@switchbot/homebridge-switchbot/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/usr/local/lib/node_modules/@switchbot/homebridge-switchbot/node_modules/axios/lib/adapters/http.js:293:11)\n    at IncomingMessage.emit (node:events:402:35)\n    at endReadableNT (node:internal/streams/readable:1343:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","config":{"transitional":{"silentJSONParsing":true,"forcedJSONParsing":true,"clarifyTimeoutError":false},"transformRequest":[null],"transformResponse":[null],"timeout":0,"xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN","maxContentLength":-1,"maxBodyLength":-1,"headers":{"Accept":"application/json, text/plain, */*","Authorization":"29c2dc18e47b5e9b74b9e87cf265fbf5dcd32a8abbdd7f693fd55986ef5f5160ff89c98f3db8ddf7747fdcc0db3c17fb","Content-Type":"application/json; charset=utf8","User-Agent":"axios/0.24.0"},"responseType":"json","method":"get","url":"https://api.switch-bot.com/v1.0/devices"},"status":401}`

**Comment by smartmatic** [comment]
Is SwitchBot down?? I am not able to get my devices through token method.

**Comment by donavanbecker** [comment]
@smartmatic, not that I am aware of. I just pushed another beta that actually may make everything work. `beta.13`

if you are still having issues, send me logs

**Comment by donavanbecker** [comment]
Version 1.2.0 has been released, Please update me on this, and if it has been resolved with this version.

**Comment by smartmatic** [comment]
I can see now the color picker in Home App but control the bulb does not work propperly. I could switch on the bulb but now the bulb stucks. The device does no longer react to any other commands.

![Bulb_01 - 1](https://user-images.githubusercontent.com/74932663/142725937-e0e48cd9-fa0e-4f43-94ab-1bc75e0085ea.jpeg)

![Bulb_02 - 1](https://user-images.githubusercontent.com/74932663/142725942-9110a829-580f-491d-9153-accb01ee3894.jpeg)

```
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb On: true
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb Brightness: 100
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb color: "0:0:0"
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb red: "0"
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb green: "0"
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb blue: "0"
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb hs: [0,0]
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb Hue: 0
[11/20/2021, 12:48:47 PM] [SwitchBot] Color Bulb: Color Bulb Saturation: 0
[11/20/2021, 12:48:47 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Brightness': characteristic was supplied illegal value: number 154 exceeded maximum of 100. See https://git.io/JtMGR for more info.

[11/20/2021, 2:37:47 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: setColorTemperature, parameter: {2146}, commandType: command
[11/20/2021, 2:37:48 PM] [SwitchBot] Color Bulb: Color Bulb Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"ExpressionAttributeValues contains invalid value: The parameter cannot be converted to a numeric value: NaN for key :dataItem"} format is invalid
[11/20/2021, 2:37:48 PM] [SwitchBot] 0
[11/20/2021, 2:37:48 PM] [SwitchBot] 0
[11/20/2021, 2:37:48 PM] [SwitchBot] Color Bulb: Color Bulb rgb: [255,204,204]
[11/20/2021, 2:37:48 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: setColor, parameter: {255}:{204}:{204}, commandType: command
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb On: true
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb Brightness: 100
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb color: "0:0:0"
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb red: "0"
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb green: "0"
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb blue: "0"
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb hs: [0,0]
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb Hue: 0
[11/20/2021, 2:37:59 PM] [SwitchBot] Color Bulb: Color Bulb Saturation: 0
[11/20/2021, 2:37:59 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Brightness': characteristic was supplied illegal value: number 466 exceeded maximum of 100. See https://git.io/JtMGR for more info.
[11/20/2021, 2:38:05 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: turnOn, parameter: default, commandType: command
[11/20/2021, 2:38:06 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: setBrightness, parameter: {100}, commandType: command
[11/20/2021, 2:38:06 PM] [SwitchBot] Color Bulb: Color Bulb Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"ExpressionAttributeValues contains invalid value: The parameter cannot be converted to a numeric value: NaN for key :dataItem"} format is invalid
[11/20/2021, 2:38:06 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: setColorTemperature, parameter: {2146}, commandType: command
[11/20/2021, 2:38:07 PM] [SwitchBot] Color Bulb: Color Bulb Device internal error due to device states not synchronized with server, Or command: {"statusCode":190,"body":{},"message":"ExpressionAttributeValues contains invalid value: The parameter cannot be converted to a numeric value: NaN for key :dataItem"} format is invalid
[11/20/2021, 2:38:07 PM] [SwitchBot] 199
[11/20/2021, 2:38:07 PM] [SwitchBot] 83
[11/20/2021, 2:38:07 PM] [SwitchBot] Color Bulb: Color Bulb rgb: [43,188,255]
[11/20/2021, 2:38:07 PM] [SwitchBot] Color Bulb: Color Bulb Sending request to SwitchBot API. command: setColor, parameter: {43}:{188}:{255}, commandType: command
```

Is node.js. 14.18.1 now really a prerequisite for version 1.2.0? 

I still have 14.18.0 installed

**Comment by donavanbecker** [comment]
@smartmatic, finally got my hands on a lightbulb and I have the latest beta closer to a finished product. Still need to tweak some things but you shouldn't be receiving anymore errors.


[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=1.2.6-beta.5/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by donavanbecker** [comment]
@smartmatic this should actually be fixed with v1.3.0

**Comment by smartmatic** [comment]
> @smartmatic this should actually be fixed with v1.3.0

@donavanbecker I can confirm that the Bulb is now working as expected. I am able to change the color with Home App. 
Only thing i observed is that the color should not change to fast. Then it looks like that the API is not able to handle the commands in the right time. Then sometimes the color don't change or it takes a while.

However thank you very much, great work! I really like that Plugin and really appreciate your effort you spend to optimize it!

**Comment by donavanbecker** [comment]
@smartmatic You could try changing the push rate of the plugin and see if that helps.
